lighthouse-viewer/app/src/lighthouse-report-viewer.js

```diff
@@ -260,6 +267,23 @@ class LighthouseViewerReport {
       // noop
     }
   }
+
+  /**
+   * Initializes of a `message` listener to respond to postMessage events.
+   */
+  listenForMessages() {
+    window.addEventListener('message', e => {
+      if (e.data.lhresults) {
```

Comment:
not a huge increase in security, but can we check the message origin/source against the opener?